### DISCUSSION TITLE
Arrows directions dynamical  vs  stochastic mode

### DISCUSSION THREAD

**Discussion by TarekGharsalli** [theislab/scvelo#726] [Q&A]
Hello,

I have a datasets 3174 (cells) * 16876 (genes).

I tried scvelo with dynamical and stochastic model, and for each model I get different arrows for some clusters.
This plot is for stochastic model: I know (with genes markers of differentiation) that arrows in clusters 6, 7 and 10 are in the best direction, but I don't have information about other clusters. 

![E10_all_cells_minshared_count_10_stochastic](https://user-images.githubusercontent.com/60474281/136412558-25b2d908-8265-4fa1-bcd1-43afb69a3ad1.png)

Then I tried the dynamical model, and I found that the arrows in cluster 10 are inverted which was good with stochastic, and also arrows are inverted in clusters 2,3, and 9.

![E10_all_cells_minshared_count_10](https://user-images.githubusercontent.com/60474281/136412580-2ee6be17-9668-4447-8a58-610b85bcdb1d.png)

I run scvelo in both model( stochastic and dynamic) with default parameters , expect  min_shared_count that I fixed it to 10 which filter out 14251 genes.

Then I tried to fixed the min_shared_count to 5 which filter out 13008 genes, and I found that the arrows in clusters 6, 7, and 10 are in the best directions but direction of arrows in clusters 2,3 and 9 still different from stochastic model.

![E10_all_cells_minshared_count_5](https://user-images.githubusercontent.com/60474281/136415167-f54aa3ee-a58b-4be1-8dca-c8d056830d41.png)

Does any one can have please can have an explanation ?

**Comment by WeilerP** [comment]
@TarekGharsalli, given the limit information, my best guess is that your phase portraits (or a majority thereof) are simply uninformative. See *e.g.* #216 or #462.

**Reply by TarekGharsalli** [reply]
Thanks for your reply @WeilerP 

I complete the analysis , and as you see in the next plot of latent time that the direction of the differentiation is from cluster 7 to cluster 6

![latent_time](https://user-images.githubusercontent.com/60474281/136535678-09806371-9e78-4468-956b-de350ea7af74.png)

I plot also Top-likelihood genes 
![genes_portrait](https://user-images.githubusercontent.com/60474281/136535926-842b958e-02ec-4f45-98e2-cd21e1f9a1f0.png)

For me  up-regulation was fitted for 995, down-regulation was fitted for 844, 712 and 835, 302 are in steady state, it's true ?
Based on the phase portraits  your confirm you hypothesis that my datasets does not provide information to infer dynamics ?
Should I use only stochastic model for this datasets or scvelo is not applicable ?

Thanks in advance :) @WeilerP

**Reply by WeilerP** [reply]
I'd say your dataset is not suitable for RNA velocity analysis. It seems to me the blue cluster is an outlier skewing your results. You do not observe a nice differentiation into different cell type along the trajectory.

**Reply by WeilerP** [reply]
Also, the latent time does not tell you anything if the data based on which is was calculated is not suitable for RNA velocity.

**Reply by TarekGharsalli** [reply]
Thanks for your reply @WeilerP 

I subset the object and I took only clusters 6, 7 and 10 which are muscles clusters, I got the arrows in the best direction but the genes portrait show that many genes up and down-regulated at the same time.

![E10_muscles_dynamique](https://user-images.githubusercontent.com/60474281/136559197-9b164783-aaf4-4a72-a688-40dd0ec953ec.png)

![genes_portrait_E10_muscles_dynamique](https://user-images.githubusercontent.com/60474281/136559200-694569d9-736d-4de8-a18c-f97c1444b1d2.png)

**Reply by WeilerP** [reply]
Again, these phase portraits are non-informative. You can infer RNA velocity when you have phase portraits showing (part of) a football/almond shape.

**Comment by TarekGharsalli** [comment]
Thanks a lot @WeilerP  for all these explanations

**Comment by TarekGharsalli** [comment]
Hello @WeilerP 
I have another question please about the phase portrait.
How many genes should be have the phase portrait football/almond shape to said that the phase portrait is informative ?
because  in the tutorial (top likelihood genes or likelihood gens per clusters ), there are some genes that also are up and down in the same time (eg Dcdc2a, Ank, Pim2, Anxa4, Rap1b) 

![phase_portrait_scvelo](https://user-images.githubusercontent.com/60474281/136771849-13107d1f-9e11-4ce3-9d8e-4536d7035081.png)

Thanks in advance @WeilerP

**Reply by WeilerP** [reply]
@TarekGharsalli, I'm not aware of a way to quantify the required number of informative genes a priori.
Sorry, not sure I correctly understand your comment about up-/down-regulation. The problem with some of the genes you mention is that their phase portraits are linear. The problem is not that up- and down-regulations are fitted but that you could also fit only a up- or down-regulation. As unsatisfying as it may be, the reason why this is not a problem here is that the models are inferred correctly in these cases. You can imagine how easily this may fail, though. You'll have to decide on a case to case basis.

**Comment by TarekGharsalli** [comment]
@WeilerP  Thanks for your reply

Can you explain more what do you mean by the models are inferred correctly ?

**Reply by WeilerP** [reply]
We know the differentiation order of the process, *i.e.*, we can visually see if an ambiguous gene is up-, down- or up- and down-regulated. I simply meant that the gene specific models match the a priori knowledge on differentiation. Hope this clarifies the statement?

**Reply by TarekGharsalli** [reply]
Thanks @WeilerP  for this clarification
